src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -627,15 +664,16 @@ def load_status_file_components(self, initial_load=False):
                         self.__configure_patching_errors = errors['details']
                         self.__configure_patching_top_level_error_count = self.__get_total_error_count_from_prev_status(errors['message'])
 
-    def __get_substatus_message(self, status_file_data, index):
-        return json.loads(status_file_data['status']['substatus'][index]['formattedMessage']['message'])
+    def __get_substatus_message(self, status_file_data, substatus_index):
+        """ Get the substatus payload message by index """
+        return json.loads(status_file_data['status']['substatus'][substatus_index]['formattedMessage']['message'])
 
     def __load_complete_status_file_data(self, file_path):
         # Read the status file - raise exception on persistent failure
         for i in range(0, Constants.MAX_FILE_OPERATION_RETRY_COUNT):
             try:
                 with self.env_layer.file_system.open(file_path, 'r') as file_handle:
-                    complete_status_file_data = json.load(file_handle)[0]    # structure is array of 1
+                    complete_status_file_data = json.load(file_handle)[0]  # structure is array of 1
```

Comment:
Marking all changes not related to this PR goal. See my other comment on what can be done with these

Reply:
this kind of change is so small, it should be overlook. lets focus on the code logic and functionalities

Reply:
Ignore this for now to look at the bigger picture. Rajasi's guidance generally applies but I'm resolving to reduce noise.